File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque
def solution(progresses, speeds):
    answer = []
    # q = deque(progresses)
    while progresses:
        cur = progresses.pop(0)
        cur_speed = speeds.pop(0)
        days = 0
        while cur < 100:
            cur += cur_speed
            days += 1
            if cur >= 100:
                tmp = 1
                while progresses:
                    if progresses[0] + speeds[0] * days >= 100:
                        progresses.pop(0)
                        speeds.pop(0)
                        tmp += 1
                    else:
                        break
                answer.append(tmp)
        
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <queue>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    queue<int> p, s;
    for (int i = 0; i < progresses.size(); i++){
        p.push(progresses[i]);
        s.push(speeds[i]);
    }
    
    vector<int> answer;
    int days = 0, cnt = 0;
    
    while (!p.empty()){
    // for (int i = 0; i < 30; i++){
        // std::cout << p.front() << " , "<< s.front() << " : " << days <<endl;

        if (p.front() + s.front() * days >= 100) {
            cnt = 0;
            while (!p.empty() and p.front() + s.front() * days >= 100){
                // if (p.front() == 0) { break; }
                std::cout << p.front() << " , "<< s.front() << " : " << days <<endl;
                p.pop();
                s.pop();
                cnt++;
            }
            // std::cout << p.front() << " , "<< s.front() << " : " << days <<endl;
            // std::cout << cnt << " : " << days <<endl;
        
            answer.push_back(cnt);
            days = 0;
            // days++;
        }
        days++;
        // else {
        //     days++;
        // }
    }
    return answer;
}